src/main/groovy/lang/ObjectRange.java

```diff
@@ -537,8 +539,9 @@ protected Object increment(Object value) {
      * @param value the value to decrement
      * @return the decremented value
      */
-    protected Object decrement(Object value) {
-        return InvokerHelper.invokeMethod(value, "previous", null);
+    @SuppressWarnings("unchecked")
+    private T decrement(T value) {
```

Comment:
I understand, please let me know what to change (if nothing else, revert problematic commits) for this to be merged.